src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -1136,7 +1145,12 @@ public override Location ParameterLocation(int index)
         public override string ParameterName(int index)
```

Comment:
>ParameterName [](start = 31, length = 13)

If we are changing this API to return null in some scenarios, consider renaming it to ```ParameterNameOrNull```. Besides more descriptive name, all the call sites will have to be changed and we can review that all of them a properly handling possible null return.

Reply:
After reviewing remaining product changes, it looks like the way ```_parameterNames``` is populated and when this API is used is not changed. It is not clear why implementation of this method is changed. Consider reverting to the original implementation.

---
In reply to: [340343304](https://github.com/dotnet/roslyn/pull/38786#discussion_r340343304) [](ancestors = 340343304)